mozilla/persona
# Usability enhancements to the forgot password screen.

- Disable the email field.
- Focus the first password field instead of the email field.
- If no email is stored (user did not come from /signin), redirect the user to /signin.
- If email address that is stored is not known, redirect to /signin.
- If email address that is stored is known but a primary, redirect to /signin
- When calling User.addressInfo on a primary address, check whether the address is known or not.  Allows us to simplify the forgot page.

This addresses both #1679 and #2130.

Additional areas to test:
- Ensure the reset password flow of the main dialog works as expected.
- Ensure the user cannot reset the password of a primary address.
- Ensure the /forgot page redirects to /signin if the email address is known.
- Ensure the /forgot page redirects to /signin if the email address is unknown.
